app/controllers/api/v1/products_controller.rb

```diff
@@ -0,0 +1,24 @@
+class Api::V1::ProductsController < Api::V1::ApiController
+  before_action :check_company_token
+  before_action :set_product, only: [:update]
+
+  def create
+    @product = Product.create!(company: @company, name: params[:name], price: params[:price])
```

Comment:
O que acha de criar associado permitindo campos?

```suggestion
    @product = @company.products.create!(product_params)
```